### PR TITLE
1.5 improvements review

### PR DESCRIPTION
**Pull request by ecino**
# 1.5 Improvements

Contains new features added after split of contracts.

## Sponsorships / Contracts

- Change of activation method to better respect the purpose of functional
  fields (is_active).
- New types of contracts : S for Sponsorships, O for Others, and SC for
                           recurring gifts.
- Cancel old invoices when a sponsorship is activated
- Call method clean_invoices not directly from workflow but from method
  contract_terminated or contract_cancelled in order to pass some context.
- Ability to set back to draft a cancelled contract.
- Generate invoices directly from a specific contract (new button).
- Generate automatically a Birthday Gift each year for Sponsorships.

## child_compassion

- Delegation until a given date : then automatically undelegates children

## crm_compassion

- Add a view for invoice lines in order to see each sponsorship paid
  given origins or ambassadors
- Abort all overrides of analytic accounts by analytic distributions in views. From
  now on, we will use only analytic accounts and will use our own rules for
  dispatching lines into several analytic accounts (with module account_analytic_attribution)
  
## message_center

- New management of gmc_states, to better track the sponsorships
- When contract is terminated, send pending gifts immediately, or abort them
  if the sponsorship is setup so (for Demaurex).
- Prevent to send LDP Gifts.
- Project suspension is no more handled by GMC State but with a Project Status
  defined in module sponsorship_tracking.
  
## partner_compassion

- Module holding all modifications of res.partner that would concern all
  compassion offices (not only Switzerland).

## Others

- Remove unused stored functional fields to improve performance
- Module remove_quick_create_partner needs only to add some global settings
  since module web_m2x_options got improved.
- Fixed some inheritance in views
- New class structure that should be used in all files for better code clarity
- A lot of bug corrections

### CODE REVIEW

File: message_center_compassion/model/gmc_message.py [hunk 1]
```diff
@@ -238,10 +238,13 @@ def reset_message(self, cr, uid, ids, context=None):
...
-            'state': 'success',
-            'failure_reason': False
-        }, context)
+        for message in self.browse(cr, uid, ids, context):
+            state = 'success'
+            if message.action_id.name == 'CreateGift':
+                state = 'fondue'
+            message.write({
+                'state': state,
+                'failure_reason': False})
```
Comment (CSester) [inline message_center_compassion/model/gmc_message.py]: No context ?

File: partner_compassion/__openerp__.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    'category': 'Partner',
+    'sequence': 150,
+    'description': """
+            Upgrade Partners to Compassion standards :
+                1. Three lines for the address
+                2. Mandatory reference sequence (CODEGA)
+                3. Church members relation between Partners
+                4. Birthdate and Deathdate
+                5. Adds new contact titles
+                6. Adds Partner Categories
```
Comment (CSester) [inline partner_compassion/__openerp__.py]: 7. Banking mandates

**Comment by CSester** [comment]
:+1: if you fix Travis :smile:

**Comment by ecino** [comment]
Travis will be fixed... small corrections made in current 1.5 branch.

**Pull request merged**